### PR TITLE
Rectified typos & grammar in docs

### PR DESCRIPTION
**Pull request by Neilblaze**
# What does this PR do?

A few typos and grammatical errors were discovered and have been corrected.
/cc: @8bitmp3, @marcvanzee 

## Checklist
- [x] This PR fixes a minor issue (e.g.: typo or small bug) or improves the docs (you can dismiss the other
      checks if that's the case).
- [x] The documentation and docstrings adhere to the
      [documentation guidelines](https://github.com/google/flax/blob/main/docs/README.md#how-to-write-code-documentation).

### CODE REVIEW

File: docs/guides/batch_norm.rst [hunk 1]
```diff
@@ -106,17 +106,7 @@ collection.
   :sync:
 
   FrozenDict({
-
```
Comment (cgarciae) [inline docs/guides/batch_norm.rst:109]: These lines are empty intentionally.

File: docs/guides/dropout.rst [hunk 2]
```diff
@@ -109,8 +109,6 @@ In short, :meth:`flax.linen.Module.make_rng` *guarantees full reproducibility*.
     def __call__(self, x):
       x = nn.Dense(self.num_neurons)(x)
 
-
```
Comment (cgarciae) [inline docs/guides/dropout.rst:112]: Same here.

**Comment by cgarciae** [comment]
Hey @Neilblaze, thanks for doing this!

There are a couple of empty lines in `.rst` files which are intentional as they align two pieces of code in a diff view.

**Comment by Neilblaze** [comment]
@cgarciae rolling back changes for those two files. I also see a couple of empty lines in [`../advanced_topics/linen_upgrade_guide.rst`](https://github.com/google/flax/blob/332b7d041de235d4579ab0fb4c82cc4ad94ade62/docs/advanced_topics/linen_upgrade_guide.rst?plain=1#L272-L279) & [`../guides/linen_upgrade_guide.rst`](https://github.com/google/flax/blob/332b7d041de235d4579ab0fb4c82cc4ad94ade62/docs/guides/linen_upgrade_guide.rst?plain=1#L272-L279) too. Should I also rollback the changes for them too?

**Comment by cgarciae** [comment]
Yeah, please rollback anything in either side of a `codediff` block.

**Review by cgarciae — APPROVED** [review]
Looks good, thanks a lot!

**Comment by IvyZX** [comment]
Hey, can you fix the pre-commit hook errors? Then I can go ahead and merge it.

**Comment by IvyZX** [comment]
Hi, could you rebase to the google/main branch and make sure all the tests pass? This is a large change and we don't want it to break head. Thank you!

**Review by 8bitmp3 — APPROVED** [review]

**Comment by Neilblaze** [comment]
@IvyZX Sure thing! I'll do it by EOD

**Comment by Neilblaze** [comment]
@IvyZX done! ✅ 
image

**Review by IvyZX — APPROVED** [review]

**Comment by chiamp** [comment]
hi @Neilblaze, seems like there's a [trailing whitespace error](https://github.com/google/flax/actions/runs/4759249733/jobs/8458402795?pr=3021)

**Pull request merged**